src/test/java/de/wejago/hichi2influxDB/unit/MqttSubscriberServiceTest.java

```diff
@@ -0,0 +1,167 @@
+        MockitoAnnotations.openMocks(this);
+        point = generateTestPoint();
+    }
+
+    @Test
+    void testConnectSuccessfully(CapturedOutput output) throws MqttException {
+        // GIVEN
+        Whitebox.setInternalState(mqttSubscriberService, "brokerIp", "localhost");
+        Whitebox.setInternalState(mqttSubscriberService, "brokerPort", "1883");
+        Whitebox.setInternalState(mqttSubscriberService, "username", "user123");
+        Whitebox.setInternalState(mqttSubscriberService, "password", "test-pass");
+
+        // WHEN
+        mqttSubscriberService.connect();
+
+        // THEN
+        assertThat(output.getOut()).contains("Successfully connected to MQTT client!");
+    }
+
+    @Test
+    void testConnectFailureOnWrongPort(CapturedOutput output) {
+        // GIVEN
+        MqttSubscriberService mqttSubscriberService = new MqttSubscriberService();
+        Whitebox.setInternalState(mqttSubscriberService, "brokerIp", "localhost");
+        Whitebox.setInternalState(mqttSubscriberService, "brokerPort", "8080");
+        Whitebox.setInternalState(mqttSubscriberService, "username", "user123");
+        Whitebox.setInternalState(mqttSubscriberService, "password", "test-pass");
+
+        // WHEN
+        assertDoesNotThrow(() -> mqttSubscriberService.connect(), "The connect() method should not throw any exceptions");
+
+        // THEN
+        assertThat(output.getOut()).contains("Error connecting to MQTT client!");
+    }
+
+    @Test
+    public void testSubscribeWhenReceivingSkippedMessage() throws Exception {
+        //GIVEN
+        String testTopic = "test/topic";
+        testSensorEntry = this.generateTestSensorEntry();
+        mqttMessage = this.generateSkippedMqttMessage();
+        when(influxDbService.writePoint(testSensorEntry)).thenReturn(point);
+        doAnswer(invocation -> {
+            IMqttMessageListener listener = invocation.getArgument(1);
+            listener.messageArrived(testTopic, mqttMessage);
+            return null;
+        }).when(mqttClient).subscribe(eq(testTopic), any(IMqttMessageListener.class));
+
+        //WHEN
+        mqttSubscriberService.subscribe(testTopic);
+
+        //THEN
+        verify(influxDbService, never()).writePoint(testSensorEntry);
+    }
+
+    @Test
+    public void testSubscribeWhenReceivingMessageToSave() throws Exception {
+        //GIVEN
+        String testTopic = "test/topic";
+        mqttMessage = this.generateMqttMessageToSave();
+
+        when(influxDbService.writePoint(testSensorEntry)).thenReturn(point);
+        doAnswer(invocation -> {
+            IMqttMessageListener listener = invocation.getArgument(1);
+            listener.messageArrived(testTopic, mqttMessage);
+            return null;
+        }).when(mqttClient).subscribe(eq(testTopic), any(IMqttMessageListener.class));
+
+        //WHEN
+        mqttSubscriberService.subscribe(testTopic);
+
+        //THEN
+        verify(influxDbService, times(1)).writePoint(any());
```

Comment:
You should somehow verify that the deserialization of json and conversion to a point works as expected.
You can also test only the small units of the conversion method. I expect JsonMapper to work as expected. ;-)

Reply:
We verify that we call the **writePoint()** method with a proper parameter. I guess this is enough, isn't it?

Reply:
It's OK